### PR TITLE
Added Checkout Started analytics event

### PR DESCRIPTION
**Pull request by Mjloturco**
Added "Checkout Started" google analytics event that fires when basket is viewed. Incudes product, basket, and discount details. 

[LEARNER-1568]

### CODE REVIEW

**Comment by Mjloturco** [comment]
@edx/helio Please review.

File: ecommerce/extensions/basket/tests/test_views.py [hunk 1]
```diff
@@ -517,6 +517,26 @@ def test_response_success(self, benefit_type, benefit_value):
             }
             calls.append(mock.call(self.site, self.user, 'Cart Viewed', properties,))
 
+            voucher = basket.vouchers.filter(id__isnull=False).first()
+            coupon = voucher.code if voucher else None
+
+            if benefit_type == Benefit.PERCENTAGE:
```
Comment (clintonb) [inline ecommerce/extensions/basket/tests/test_views.py:523]: What is this for? The discount should always be the dollar amount.

File: ecommerce/extensions/basket/tests/test_views.py [hunk 2]
```diff
@@ -517,6 +517,26 @@ def test_response_success(self, benefit_type, benefit_value):
...
+                discount = (basket.total_incl_tax_excl_discounts * benefit_value / 100)
+            else:
+                discount = benefit_value
+            revenue = basket.total_incl_tax_excl_discounts - discount
+            properties = {
+                'order_id': basket.order_number,
+                'value': str(basket.total_incl_tax_excl_discounts),
+                'revenue': str(revenue),
```
Comment (clintonb) [inline ecommerce/extensions/basket/tests/test_views.py:531]: `basket.total_excl_tax`

File: ecommerce/extensions/basket/tests/test_views.py [hunk 3]
```diff
@@ -517,6 +517,26 @@ def test_response_success(self, benefit_type, benefit_value):
...
+            coupon = voucher.code if voucher else None
+
+            if benefit_type == Benefit.PERCENTAGE:
+                discount = (basket.total_incl_tax_excl_discounts * benefit_value / 100)
+            else:
+                discount = benefit_value
+            revenue = basket.total_incl_tax_excl_discounts - discount
+            properties = {
+                'order_id': basket.order_number,
+                'value': str(basket.total_incl_tax_excl_discounts),
```
Comment (clintonb) [inline ecommerce/extensions/basket/tests/test_views.py:530]: `basket.total_excl_tax_excl_discounts`

**Review by clintonb — CHANGES_REQUESTED** [review]

**Pull request closed without merging**